pkg/receive/handler.go

```diff
@@ -81,41 +72,56 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 	}
 
 	h := &Handler{
-		logger:       logger,
-		router:       router,
-		readyStorage: o.ReadyStorage,
-		receiver:     o.Receiver,
-		options:      o,
-		quitCh:       make(chan struct{}),
+		logger:          logger,
+		router:          router,
+		readyStorage:    o.ReadyStorage,
+		receiver:        o.Receiver,
+		options:         o,
+		quitCh:          make(chan struct{}),
+		readinessProber: o.ReadinessProber,
 	}
 
 	readyf := h.testReady
 	router.Post("/api/v1/receive", readyf(h.receive))
+	o.Registry.MustRegister(
+		requestDuration,
+		responseSize,
+	)
 
 	return h
 }
 
+// Ready sets Handler to be healthy.
+func (h *Handler) Healthy() {
+	h.readinessProber.SetHealthy()
+}
+
 // Ready sets Handler to be ready.
 func (h *Handler) Ready() {
-	atomic.StoreUint32(&h.ready, 1)
+	h.readinessProber.SetReady()
 }
 
 // Verifies whether the server is ready or not.
-func (h *Handler) isReady() bool {
-	ready := atomic.LoadUint32(&h.ready)
-	return ready > 0
+func (h *Handler) isReady() error {
+	return h.readinessProber.IsReady()
+}
+
+// HandleInMux hadles this router in specified mux on given part
+func (h *Handler) HandleInMux(path string, mux *http.ServeMux) {
+	mux.Handle(path, h.router)
 }
 
 // Checks if server is ready, calls f if it is, returns 503 if it is not.
 func (h *Handler) testReady(f http.HandlerFunc) http.HandlerFunc {
```

Comment:
Should this be part of the prober package?

Reply:
Yes, thanks I must have overlooked this somehow. 
Moved to prober package.